src/network/networkd-dhcp6.c

```diff
@@ -251,6 +252,12 @@ int dhcp6_configure(Link *link) {
         if (r < 0)
                 goto error;
 
+        if (link->network->rapid_commit) {
+                r = sd_dhcp6_client_set_request_option(client, SD_DHCP6_OPTION_RAPID_COMMIT);
+                if (r < 0)
+                        goto error;
+        }
```

Comment:
hmm, is it really sufficient to just request this option? I mean, why don't we have to make changes to the state engine of the dhcp client, too?

I didn't read up on dhcp rapid commit, but it appears to be something that requires more client side changes than just requesting it.

Reply:
The client part is handled as per https://tools.ietf.org/html/rfc3315#section-17.1.1  `DHCP6_STATE_SOLICITATION` in client_receive_advertise.

Reply:
so, what are you saying? we already implement that part correctly?

Reply:
Well I think so. Since enabling this in the request we can see the change in packet exchange.
I have attached the screenshorts.

![Uploading Screenshot from 2017-11-27 17-48-46.png…]()
![screenshot from 2017-11-27 17-51-19](https://user-images.githubusercontent.com/145210/33266474-d2e54734-d39b-11e7-98d4-4ae3308c4061.png)

Reply:
![screenshot from 2017-11-27 17-48-46](https://user-images.githubusercontent.com/145210/33266529-086c3020-d39c-11e7-93bc-b4f06b6ab129.png)